File: app/src/main/assets/scenes/fps_calculator.js
function FpsCalculator() {

    this.MAX_BUFFER_SIZE = 100;

    this.buffer = [];

    this.fps = 0;

    this.update = function(dt) {
        this.buffer.push(dt);
        if (this.buffer.length > this.MAX_BUFFER_SIZE) {
            this.buffer.shift();
        }

        this.fps = 0;
        var that = this;
        this.buffer.forEach(function(item, index, array) {
            that.fps += item;
        });
        this.fps /= this.buffer.length;
        this.fps = 1 / this.fps;
    };
}

File: app/src/main/java/ilapin/opengl_research/domain/FpsCalculator.kt
package ilapin.opengl_research.domain

import java.util.*

/**
 * @author Игорь on 02.03.2020.
 */
class FpsCalculator {

    private var _fps = 0f

    private val buffer = LinkedList<Float>()

    val fps: Float
        get() = _fps

    fun update(dt: Float) {
        buffer.addLast(dt)
        if (buffer.size > MAX_BUFFER_SIZE) {
            buffer.removeFirst()
        }

        _fps = buffer.sum() / buffer.size
        _fps = 1 / _fps
    }

    companion object {

        private const val MAX_BUFFER_SIZE = 100
    }
}